### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by panianchuk**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,21 @@
...
+      for (const i of action.keysToRemove) {
+        delete state[i];
+      }
+    } else if (action.type === 'clear') {
+      for (const key in state) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
+      }
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/transformState.js:21]: avoid such nesting `else if` conditions
This code is hard to read and maintain
you can use `switch case` here

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { keysToRemove, extraData, type } = action;
+
+    switch (true) {
+      case type === 'addProperties':
```
Comment (Bohdan-Maksymiuk) [inline src/transformState.js:12]: same comment
[usefulLink](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)
```suggestion
    switch (type) {
      case 'addProperties':
```
fix everywhere

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,29 @@
...
+        Object.assign(state, extraData);
+        break;
+
+      case type === 'removeProperties':
+        for (const i of keysToRemove) {
```
Comment (Bohdan-Maksymiuk) [inline src/transformState.js:17]:
```suggestion
        for (const key of keysToRemove) {
```
use meaningful names

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/transformState.js (hunk 2 above)
Comment (loralevitska) [reply src/transformState.js:12]: not fixed. If your have some problems with deploy fill free to ask in fe_chat

**Review by loralevitska — CHANGES_REQUESTED** [review]
hi! did you push your changes to github? Previous comments didn't fix.

**Review by Viktor-Kost — APPROVED** [review]

**Pull request closed without merging**